### PR TITLE
Correctly generate the edit URL of a version

### LINKED ISSUE #4460: Dashboard version compare does not work
### Affected version(s)

4.13.2

### Description

I noticed today, when i click on show differences, the window shows the dashboard again
grafik
grafik

after closing and reloading, another version was created, and so on.

grafik

I have this on 2 installations, I think that's not right.

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #4460

This is because the icon now uses the new `toggle()` method.

### CODE REVIEW

File: core-bundle/src/Resources/contao/classes/Versions.php [hunk 1]
```diff
@@ -759,32 +759,36 @@ protected function getEditUrl()
 			return sprintf($this->strEditUrl, $this->intPid);
 		}
 
-		$strUrl = Environment::get('request');
+		$pairs = array();
+		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
 
-		// Save the real edit URL if the visibility is toggled via Ajax
-		if (preg_match('/&(amp;)?state=/', $strUrl))
+		parse_str($request->server->get('QUERY_STRING'), $pairs);
```
Comment (ausi) [inline core-bundle/src/Resources/contao/classes/Versions.php:765]: Is there a specific reason we don’t use `$request->query->all()` here?

**Review by ausi — APPROVED** [review]

File: core-bundle/src/Resources/contao/classes/Versions.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/Versions.php:765]: Good point. We are doing it this way here:

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/library/Contao/Controller.php#L1119

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/modules/ModuleLogout.php#L73

https://github.com/contao/contao/blob/ab71fd99f6c7ce36595a0d8dd75b17710d98a2bb/core-bundle/src/Resources/contao/pages/PageLogout.php#L62

But maybe we should use `$request->query->all()` everywhere instead?

Comment (ausi) [reply core-bundle/src/Resources/contao/classes/Versions.php:765]: But not in a bugfix version then I think. I’d merge this as is and maybe change it in Contao 5.

Comment (Toflar) [reply core-bundle/src/Resources/contao/classes/Versions.php:765]: Yeah that looks really weird but I agree changing it in 5.

**Review by Toflar — APPROVED** [review]

**Pull request merged**